File: app/src/main/java/com/example/seminariodelenguajeskotlin/AppDataBase.java
package com.example.seminariodelenguajeskotlin;


import androidx.room.Database;
/*
@Database(entities = [Personaje::class], version = 1)

abstract class AppDataBase:RoomDatabase() {

    abstract fun personajeDao() : PersonajeDao
    companion object{
            private var INSTANCIA : AppDataBase? = null

            fun getDataBase(contexto:Contexto):AppDataBase{
            if(INSTANCIA==NULL ){
                synchronized(this)
                INSTANCIA = Room.databaseBuilder(
                context,
                AppDatabase::class.java, "personajes_database")
                .allowMainThreadQueries()
                .fallbackToDestructiveMigration()
                .build()
            }
        }


    }
}
*/

File: app/src/main/java/com/example/seminariodelenguajeskotlin/AppDataBase.kt
package com.example.seminariodelenguajeskotlin


import android.content.Context
import androidx.room.Database;
import androidx.room.Room
import androidx.room.RoomDatabase



//base de datos
@Database(entities = [Personaje::class], version = 1)
abstract class AppDataBase:RoomDatabase() {
    abstract fun personajeDao(): PersonajeDao

    companion object {
        private var INSTANCIA: AppDataBase? = null

        fun getDataBase(contexto: Context): AppDataBase {
            if (INSTANCIA == null) {
                synchronized(this) {
                    INSTANCIA = Room.databaseBuilder(
                        contexto, AppDataBase::class.java, "personajes_database"
                    )
                        .allowMainThreadQueries()
                        .fallbackToDestructiveMigration()
                        .build()
                }
            }
            return INSTANCIA!!
        }
    }
}
